### ISSUE TITLE
Improve the Gallery images alt attribute within the editor block

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#16886]
**Describe the bug**
Gallery images in Gallery block _within the editor_ are typically rendered with the following markup (new lines are for better readability in this example):

```
<img
	src=" ... "
	alt="Huatulco Coastline"
	data-id="770"
	tabindex="0"
	aria-label="image 3 of 4 in gallery"
>
```

The `aria-label="image 1 of 4 in gallery"` overrides the `alt` attribute, making the images not recognizable by screen reader users. The only available information is `image 3 of 4 in gallery`, which doesn't help that much.

**To reproduce**
- create a post
- insert a gallery containing images with alt attributes
- check the source in your browser's dev tools
- see both the alt attribute and the aria-label attribute are present
- optionally: try to use a screen reader and navigate to a gallery image within the editor (that's not easy)
- verify the images are announced only with the aria-label content, e.g. `image 3 of 4 in gallery`

**Expected behavior**
The alt attribute to not be overridden

**Desktop (please complete the following information):**
N/A

**Smartphone (please complete the following information):**
N/A

**Possible solution:**
- Prepend the alt attribute (if any) to the aria-label
- To be verified: whether the info `image 3 of 4 in gallery` is really necessary: gallery images are within list item and the number of items should be automatically announced by screen readers

**Comment by paaljoachim** [comment]
@glendaviesnz Glen

As your working on https://github.com/WordPress/gutenberg/pull/25940 
I do believe this is an issue you might want to know about.

**Comment by glendaviesnz** [comment]
With the merging of the [gallery block refactor](https://github.com/WordPress/gutenberg/pull/25940) the gallery images are now Image blocks, which don't currently have an aria label so alt tag should be read by default.
